tests/js/set-tests.js

```diff
@@ -0,0 +1,392 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+'use strict';
+const AppConfig = require("./support/testConfig");
+
+// Prevent React Native packager from seeing modules required with this
+const require_method = require;
+function nodeRequire(module) {
+    return require_method(module);
+}
+
+function closeAfterUpload(realm) {
+    return realm.syncSession.uploadAllLocalChanges().then(() => realm.close());
+}
+
+const Realm = require('realm');
+const TestCase = require('./asserts');
+const schemas = require('./schemas');
+const Utils = require('./test-utils');
+const { Decimal128, ObjectId } = require("bson");
+//const { Test } = require('tslint');
+
+let pathSeparator = '/';
+const isNodeProcess = typeof process === 'object' && process + '' === '[object process]';
+const isElectronProcess = typeof process === 'object' && process.versions && process.versions.electron;
+
+if (isNodeProcess && process.platform === 'win32') {
+    pathSeparator = '\\';
+}
+
+const fs = isNodeProcess ? nodeRequire('fs-extra') : require('react-native-fs');
+
+module.exports = {
+    async testSetSchema() {
+        // test that short (JS) and canonical schema types yield
+        // the same results
+        const shorthand_schema = {
+            name: "ShorthandSchema",
+            properties: {
+                Names: "string<>"
+            }
+        };
+
+        const canonical_schema = {
+            name: "CanonicalSchema",
+            properties: {
+                Names:  {type: 'set',   objectType: 'string'}
+
+            }
+        };
+
+        const shorthand_realm = new Realm({ schema: [shorthand_schema] });
+        const sh_schema = shorthand_realm.schema;
+        shorthand_realm.close();
+
+        const canonical_realm = new Realm({ schema: [canonical_schema] });
+        const can_schema = canonical_realm.schema;
+        canonical_realm.close();
+
+        TestCase.assertEqual(sh_schema.properties, can_schema.properties, 
+            "Canonical and shorthand schemas should have idendical properties");
```

Comment:
Personally I wouldn't bother with these messages, the assertion library should clearly state the issue and jumping to the offending line would make it obvious what is going on. On the other hand, when test cases are copy-pasted (as they often are) I've seen very frequently that these messages are not updated to match the code and they become more of a liability than a help to the developer.

Again - this is a personal preference, feel free to disagree 🙂

Reply:
I like having the messages there.  For example, if you test that the size of `teams` is on several occasions, Jamine's output won't tell you which of the tests has failed.  Adding different messages will help locate the failing test.

Reply:
I find them useful when asserting in a loop - good to know which iteration is failing.